core/src/banking_stage/transaction_scheduler/transaction_state_container.rs

```diff
@@ -81,27 +79,35 @@ pub(crate) trait StateContainer<Tx: TransactionWithMeta> {
             .expect("transaction must exist");
         let priority_id = TransactionPriorityId::new(transaction_state.priority(), transaction_id);
         transaction_state.transition_to_unprocessed(transaction_ttl);
-        self.push_id_into_queue(priority_id);
+        self.push_ids_into_queue(std::iter::once(priority_id));
     }
 
-    /// Pushes a transaction id into the priority queue. If the queue is full, the lowest priority
-    /// transaction will be dropped (removed from the queue and map).
-    /// Returns `true` if a packet was dropped due to capacity limits.
-    fn push_id_into_queue(&mut self, priority_id: TransactionPriorityId) -> bool;
+    /// Pushes transaction ids into the priority queue. If the queue if full,
+    /// the lowest priority transactions will be dropped (removed from the
+    /// queue and map) **after** all ids have been pushed.
+    /// To avoid allocating, the caller should not push more than
+    /// [`EXTRA_CAPACITY`] ids in a call.
+    /// Returns the number of dropped transactions.
+    fn push_ids_into_queue(
+        &mut self,
+        priority_ids: impl Iterator<Item = TransactionPriorityId>,
+    ) -> usize;
```

Comment:
i think we could've got away without some of this refactor stuff for the bp-targeted changes

Reply:
This was necessary, lots of context in the original pr

Reply:
Yeah was originally not targeting it for a backport, just because of other stuff it got delayed until after 2.2 was cut